server/db/actions/Dog.js

```diff
@@ -0,0 +1,22 @@
+import dbConnect from "../dbConnect";
+import Dog from "../models/Dog";
+
+export async function getDogs(filter = {}) {
+  await dbConnect();
+
+  try {
+    return Dog.find(filter);
```

Comment:
- for `name` it would be nice to use LIKE rather than strict equals (see ex https://mongoosejs.com/docs/api/model.html#example)
- for `instructors` we don't want element order to matter and we should use the `$contains` operator, so any dog object with _either_ instructor A or instructor B can be returned